reframe/core/schedulers/slurm.py

```diff
@@ -91,8 +91,35 @@ def __init__(self):
         self._update_state_count = 0
         self._completion_time = None
 
-    @property
-    def completion_time(self):
+    def completion_time(self, job):
+        if (self._completion_time or
+            not slurm_state_completed(job.state)):
+            return self._completion_time
+
+        slurm_variable = [('SLURM_TIME_FORMAT', 'standard')]
+        with env.temp_environment(variables=slurm_variable):
+            completed = _run_strict(
+                'sacct -S %s -P -j %s -o jobid,end' %
+                (datetime.now().strftime('%F'), job.jobid),
+                log=False
+            )
+
+        # This matches the format for both normal jobs as well as job arrays.
+        # For job arrays the job_id has one of the following formats:
+        #   * <job_id>_<array_task_id>
+        #   * <job_id>_[<array_task_id_start>-<array_task_id_end>]
+        # See (`Job Array Support<https://slurm.schedmd.com/job_array.html`__)
```

Comment:
Don't copy paste the comment. I would define the regex at the class level or at least this basic pattern that matches the job id and re use that both here and in `_update_state`. And I'd move the comment there as well. In that hoisted regex pattern, don't define any capturing group, such that you can write it as follows:

```python
class SlurmJobScheduler:
    # Comment from Theo
    _state_patt = r'\d+(?:_\d+|_\[\d+-\d+\])?'

    def completion_time(self, job):
        ...
        state_match = re.findall(r'^(?<jobid>%s)\|(?P<end>\S+)' % self._state_patt, ...)
```